siliconcompiler/design.py

```diff
@@ -0,0 +1,535 @@
+import re
+from pathlib import Path
+from typing import List
+from siliconcompiler.schema import NamedSchema
+from siliconcompiler.schema import EditableSchema, Parameter, Scope
+from siliconcompiler.schema.utils import trim
+from siliconcompiler import utils
+from siliconcompiler import SiliconCompilerError
+
+
+###########################################################################
+class DesignSchema(NamedSchema):
+
+    def __init__(self, name: str):
+        NamedSchema.__init__(self, name=name)
+        schema_design(self)
+        # TODO: move this into use schema?
+        self.__dependency = {}
+
+    ############################################
+    def set_topmodule(self, fileset: str, value: str) -> str:
+        """Sets topmodule for a fileset.
+
+        Args:
+           fileset (str): Fileset name.
+           value (str): Topmodule name.
+
+        Returns:
+           str: Topmodule name
+
+        Notes:
+        - first character must be letter or underscore
+        - remaining characters can be letters, digits, or underscores
```

Comment:
why particular reason for this restriction?

Reply:
is there any concern this will not be applicable to HLS and any other languages?